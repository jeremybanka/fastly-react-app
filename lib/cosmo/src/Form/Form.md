```jsx
<Form onSubmit={() => console.log("Form submitted")}>
  <Form.Header>
    <Form.Title>Form title</Form.Title>
    <Form.Subtitle>Form subtitle</Form.Subtitle>
    <Form.Body>
      <Form.Section>[some inputs here]</Form.Section>
      <Form.Section>[more inputs here]</Form.Section>
    </Form.Body>
    <Form.Actions>[buttons here]</Form.Actions>
  </Form.Header>
</Form>
```
