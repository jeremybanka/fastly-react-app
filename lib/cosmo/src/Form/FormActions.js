/* @flow */

import * as React from "react";
import { Box } from "../";

type Props = {
  children: React.Node,
};

const FormActions = (props: Props) => (
  <Box borderTop="light" paddingTop="md" marginTop="md">
    {props.children}
  </Box>
);

export default FormActions;
