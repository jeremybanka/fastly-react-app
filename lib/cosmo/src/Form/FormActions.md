The `<Form.Actions />` component should be a direct child of `<Form />`.
