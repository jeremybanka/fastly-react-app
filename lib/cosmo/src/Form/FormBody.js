/* @flow */

import * as React from "react";
import { Box } from "../";

type Props = { children: React.Node };

const FormBody = (props: Props) => <Box>{props.children}</Box>;

export default FormBody;
