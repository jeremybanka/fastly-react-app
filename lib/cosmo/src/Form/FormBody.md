The `<Form.Body />` component should be a direct child of `<Form />`.
