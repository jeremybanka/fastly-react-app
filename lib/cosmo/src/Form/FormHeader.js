/* @flow */

import * as React from "react";
import { Box } from "../";

type Props = { children: React.Node };

const FormHeader = (props: Props) => <Box>{props.children}</Box>;

export default FormHeader;
