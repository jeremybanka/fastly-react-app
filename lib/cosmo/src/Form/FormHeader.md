The `<Form.Header />` component should be a direct child of `<Form />`.
