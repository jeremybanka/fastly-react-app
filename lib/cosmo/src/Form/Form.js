/* @flow */

import * as React from "react";
import { Box } from "../";
import FormActions from "./FormActions";
import FormBody from "./FormBody";
import FormHeader from "./FormHeader";
import FormSection from "./FormSection";
import FormTitle from "./FormTitle";
import FormSubtitle from "./FormSubtitle";

type Props = {
  children: React.Node,
  onSubmit: (any) => void | Promise<*>,
};

const Form = (props: Props) => {
  // this smells
  const handleSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
    event.preventDefault();
    props.onSubmit(event);
  };

  return (
    <Box as="form" onSubmit={handleSubmit}>
      {props.children}
    </Box>
  );
};

Form.Actions = FormActions; // rename Form.Footer?
Form.Body = FormBody;
Form.Header = FormHeader;
Form.Section = FormSection; // define intended usage
Form.Title = FormTitle;
Form.Subtitle = FormSubtitle; // candidate for removal?

export default Form;
