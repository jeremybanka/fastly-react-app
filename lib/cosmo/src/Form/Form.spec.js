/* @flow */

import React from "react";
import { render } from "enzyme";
import Form from "./Form";

describe("Form test suite", () => {
  it("renders the stored snapshot", () => {
    const handleSubmit = jest.fn();
    const component = <Form onSubmit={handleSubmit}>Hi</Form>;
    expect(render(component)).toMatchSnapshot();
  });
});
