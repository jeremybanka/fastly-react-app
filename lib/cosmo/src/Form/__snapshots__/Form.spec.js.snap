// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test suite renders the stored snapshot 1`] = `
<form
  class="Box-sc-1v2etov-0 kAhoak"
>
  Hi
</form>
`;
